lib/Sabberworm/CSS/Value/Size.php

```diff
@@ -36,13 +37,39 @@ public static function parse(ParserState $oParserState, $bIsColorComponent = fal
 			}
 		}
 
-		$sUnit = null;
-		$aSizeUnits = self::getSizeUnits();
-		foreach($aSizeUnits as $iLength => &$aValues) {
+        $sParsedUnit = '';
+        $iOffset = 0;
+		while (true) {
+		    $sChar = $oParserState->peek(1, $iOffset);
+            $iPeek = ord($sChar);
+
+            // Ranges: a-z A-Z 0-9 %
+            if (($iPeek >= 97 && $iPeek <= 122) ||
+                ($iPeek >= 65 && $iPeek <= 90) ||
+                ($iPeek >= 48 && $iPeek <= 57) ||
+                ($iPeek === 37)) {
+                $sParsedUnit .= $sChar;
+                $iOffset++;
+            } else {
+                break;
+            }
+        }
+
+        $sUnit = null;
+        $aSizeUnits = self::getSizeUnits();
+
+		foreach($aSizeUnits as $iLength => $aValues) {
+		    $iConsumeLength = $iLength;
 			$sKey = strtolower($oParserState->peek($iLength));
 			if(array_key_exists($sKey, $aValues)) {
+			    if (strtolower($sParsedUnit) !== $sKey) {
+                    if (!$oParserState->getSettings()->bLenientParsing) {
+                        throw new UnexpectedTokenException('Unit', $sParsedUnit, 'identifier', $oParserState->currentLine());
+                    }
+                    $iConsumeLength = strlen($sParsedUnit);
+                }
```

Comment:
This appears to be using spaces for indentation rather than tabs.

Reply:
Addressed in https://github.com/sabberworm/PHP-CSS-Parser/pull/193/commits/54ae4fe617dafc6771db717cf85d986d4300275a.